lib/sqlalchemy/sql/selectable.py

```diff
@@ -3643,11 +3670,11 @@ class SelectStatementGrouping(GroupedElement, SelectBase):
 
     element: SelectBase
 
-    def __init__(self, element):
+    def __init__(self, element: Select[Any]) -> None:
         self.element = coercions.expect(roles.SelectStatementRole, element)
 
-    def _ensure_disambiguated_names(self):
-        new_element = self.element._ensure_disambiguated_names()
+    def _ensure_disambiguated_names(self) -> SelectStatementGrouping:
```

Comment:
**mike bayer** (zzzeek) wrote:

remove

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4325